TROUBLESHOOT.md

```diff
@@ -0,0 +1,165 @@
+BigQuery Schema:
+#### For Datastream API
+##### Case 1: The Record's Schema is incompatible with the BigQuery Table Schema
+- <b>Record value matches the expected BQ schema type</b> (but does not follow the schema rules)
+    - <i>Example: Read from a table having field of type `STRING` and write to BQ table having 
+field of type `INTEGER`, but the record value is modified from the read string input to a 
+long value.</i>
+  - This case works without any problem since the serializer matches the expected BQ Schema type
+to the passed record value.
+- <b>Record value matches the record schema type (but incompatible with BQ Table Schema)
+<br>
+OR
+<br>
+- Record value does not match either schema type </b>
+  - This is a mismatch between the expected BQ Schema type and the passed value. 
+So, `BigQueryConnectorException` would be thrown indicating that the pipeline is unable to 
+forward the record to the next operator (sink).
+##### Case 2: The Record's Schema is compatible with BigQuery Table Schema
+- <b> Record value matches the record schema type and the expected BQ schema type </b>
+
+  - The desired case, no issues.
+- <b> Record value does not match either schema type </b>
+
+  - This is a mismatch between the expected BQ Schema type and the passed value. 
+So, `BigQueryConnectorException` would be thrown indicating that the pipeline is unable to 
+forward the record to the next operator (sink).
+
+####  For Table API
+##### Case 1: The record schema is incompatible with the BigQuery Table Schema
+- The schema of the source (transformed source table) should match the schema of the sink table 
+(BQ Table).
+- In case of a mismatch `org.apache.flink.table.api.ValidationException` is thrown.
+##### Case 2: The records schema is compatible with BigQuery table schema
+- <b> Record value matches the record schema type and the expected BQ schema type </b>
+  
+  - The desired case, no issues.
+- <b> Record value does not match either schema type </b>
+
+  - `org.apache.flink.table.api.ValidationException` is thrown when the RowData values do not match 
+the described schema.
+
+### Records are not available in the BigQuery Table when checkpointing is disabled.
+- The connector relies on checkpoints for triggering writes to the BigQuery table.
+- It is important to enable checkpointing and configure the interval suitably to ensure smooth 
+functioning of the connector.
+
+### Problems related to unbounded reads
+- Users could be facing problems involving:
+    - Skipping of "completed" partitions. The connector is unable to re-read the partitions if it 
+has read partitions with greater timestamp after the previous read. This is because the connector 
+marks the partitions as "completed" and does not re-read them. 
+- The connector team is working on improving the unbounded read and to provide improved 
+functionality, previously supported unbounded read is going to be removed soon. 
+Improved unbounded read would be provided soon.
```

Comment:
Just mention that it does not work correctly and must not be used. Team will remove this feature soon, and rebuild from scratch in 2025.

Reply:
Done.